src/utils/elasticDSL.js

```diff
@@ -94,6 +94,10 @@ let basicSimplifyTree = _.flow(
   }
 )
 
+let flatCompact = _.flow(_.flatten, _.compact)
+let unlessEmpty = onFalse =>
```

Comment:
What should `unlessEmpty` return if its argument is in fact empty? Will this cause malformed elastic queries?

Reply:
It will return that empty/falsy argument as specified in `F.unless`. It should not do any harm as the output of this function is spread or merged.
But I think it's not ideal when input is `[]`. It will output `[]` and spread `length: 0` on the object. So I will make it return `false` explicitly